a11y review and functional refactor

blackfalcon opened the pull request:
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Resolves:** #64, #55, #66

## Summary:

_Please summarize the scope of the changes you have submitted, what the intent of the work is and anything that describes the before/after state of the project._

The scope of this PR will address the manual assignment of an index attribute to manage keyboard events and pre-selected options. 

This PR also addresses the remaining a11y interactions and voice-over feedback. 

**Demo:** https://alaskaairlines-auro-menu-pr-81.surge.sh/

## Type of change:

Please delete options that are not relevant.

- [ ] New capability
- [x] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)


## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

jason-capsule42 commented on src/auro-menu.js:
```diff
@@ -45,126 +46,149 @@ class AuroMenu extends LitElement {
+    if (!event.target.disabled) {
+      this.items.forEach((item) => {
+        item.tabIndex = -1;
+        item.removeAttribute('selected');
```
If we removed the selected state when we disable a menu option then we need to dispatch the event as well so that parents like dropdownmenu can respond to the de-selection.

blackfalcon replied:
https://github.com/AlaskaAirlines/auro-menu/blob/4f1a8f2277b2d6a9345ed3f1438d8a8220cf328e/src/auro-menu.js#L52-L68

The intent of this loop is on `click` the all other previously selected items would be returned to a `0` state. The second part is the application of the new value and state. 

In the next iteration, I will be looking at emitting additional events as to alert the consumer that the value did change.

blackfalcon replied:
#82 was created to address this feedback.

blackfalcon committed:
feat(a11y): increase interactive keyboard experience #55 #64

This commit removes the manual complexity with keyboard experiences
and adds support for aria roles to increase voice over experiences.

Changes to be committed:
modified:   src/auro-menu.js
modified:   src/auro-menuoption.js
modified:   src/style-menuoption.scss
modified:   test/auro-menu.test.js
modified:   web-test-runner.config.mjs

blackfalcon committed:
docs(*): update demo examples, api and readme docs

Changes to be committed:
modified:   README.md
modified:   demo/demo.md
modified:   docs/api.md

blackfalcon committed:
docs(api): remove @private reference over public api attrs

Changes to be committed:
modified:   docs/api.md
modified:   src/auro-menuoption.js

jason-capsule42 commented:
Sorry for adding this late, thought for sure I already put this in here.

After linking this to dropdownmenu locally, updating the eventListener to the new name and accounting for the change away from using index to make a selection I noticed the following regressions:

1. The selectedOption event is fired when first loading the menu even when no value is pre-selected.
2. Making a selection manually after load is not firing the selectedOption event
3. When the selectedOption event is fired it doesn't it is not sending event.target.value or displayValue. Both are necessary for dropdownmenu since dropdownmenu needs the displayValue to put into the trigger.

blackfalcon commented:
@jason-capsule42 dropdown menu has a dependency on menu, not the other way around. The scope of this PR is to refactor the interaction functions of menu to meet an a11y spec that the previous version did not. 

After this merge, dropdown menu (which is not released) will need to be refactored.
